src/ai/backend/manager/models/agent.py

```diff
@@ -142,6 +142,20 @@ async def get_schedulable_agents_by_sgroup(
             db=db,
         )
 
+    async def get_occupied_slots(self, db: ExtendedAsyncSAEngine) -> ResourceSlot:
+        async with db.begin_readonly_session() as db_session:
+            query = sa.select(KernelRow.occupied_slots).where(
+                sa.and_(
+                    KernelRow.agent == self.id,
+                    KernelRow.status.in_(AGENT_RESOURCE_OCCUPYING_KERNEL_STATUSES),
+                )
+            )
+            kernel_slots = (await db_session.scalars(query)).all()
+            occupied_slots = ResourceSlot()
+            for slots in kernel_slots:
+                occupied_slots += slots
+            return occupied_slots
```

Comment:
It doesn't look great to pass the db as an argument, but I'll apply it as is for now.
However, since occupied slots are only created when a container is actually running on the agent, it seems appropriate to use requested_slots and occupied_slots depending on the status.

Reply:
It seems like occupied_slots only needs to be checked in the RUNNING status.